src/kibali/AuthZChecker.cs

```diff
@@ -133,38 +133,61 @@ private HashSet<PermissionsError> InvertPermissionsDocument(PermissionsDocument
                             resources.Add(pathKey, resource);
                         }
                         var leastPrivilegedPermissionSchemes = ParseLeastPrivilegeSchemes(path.Value);
-                        resource.AddRequiredClaims(permission.Key, pathSet, leastPrivilegedPermissionSchemes, provisioningData);
+                        var alsoRequires = ParseAlsoRequiresPermissions(path.Value);
+                        resource.AddRequiredClaims(permission.Key, pathSet, leastPrivilegedPermissionSchemes, provisioningData, alsoRequires);
                         if (validate)
                         {
-                            errors.UnionWith(resource.ValidateLeastPrivilegePermissions(permission.Key, pathSet, leastPrivilegedPermissionSchemes));
+                            foreach (var requiredPermission in alsoRequires)
+                            {
+                                if (!permissionsDocument.Permissions.TryGetValue(requiredPermission, out Permission value))
+                                {
+                                    errors.Add(new PermissionsError { ErrorCode = PermissionsErrorCode.InvalidAlsoRequiresPermission, Message = $"{pathKey} which has permission {permission.Key} also requires permissions that don't exist - {requiredPermission}", Path = pathKey });
+                                }
+                                else
+                                {
+                                    var permissionPathSetSchemes = value.Schemes.Keys;
+                                    var invalidAlsoRequiresSchemes = pathsetSchemes.Where(c => !permissionPathSetSchemes.Contains(c));
+                                    if (invalidAlsoRequiresSchemes.Any())
```

Comment:
```suggestion
                                    if (pathsetSchemes.Any(c => !permissionPathSetSchemes.Contains(c)))
```

Reply:
We need the variable for the error message.